src/VisualStudio/Next/Remote/ServiceHubRemoteHostClient.JsonRpcSession.cs

```diff
@@ -160,6 +154,62 @@ private class SnapshotJsonRpcClient : JsonRpcClient
                     }
                 }
 
+                private async Task WriteAssetAsync(ObjectWriter writer, byte[][] checksums)
+                {
+                    // special case
+                    if (checksums.Length == 0)
+                    {
+                        await WriteNoAssetAsync(writer).ConfigureAwait(false);
+                        return;
+                    }
+
+                    if (checksums.Length == 1)
+                    {
+                        await WriteOneAssetAsync(writer, checksums[0]).ConfigureAwait(false);
+                        return;
+                    }
+
+                    await WriteMultipleAssetsAsync(writer, checksums).ConfigureAwait(false);
+                }
+
+                private Task WriteNoAssetAsync(ObjectWriter writer)
+                {
+                    writer.WriteInt32(0);
+                    return SpecializedTasks.EmptyTask;
+                }
+
+                private async Task WriteOneAssetAsync(ObjectWriter writer, byte[] checksum)
+                {
+                    var service = ChecksumScope.Workspace.Services.GetRequiredService<ISolutionChecksumService>();
+
+                    var checksumObject = service.GetChecksumObject(new Checksum(checksum), _source.Token);
+                    writer.WriteInt32(1);
+
+                    writer.WriteValue(checksum);
```

Comment:
here you wrote the checksum as a value.  BElow you wrote the checksum as an array.  Why the difference?